crypto/evp/e_aes.c

```diff
@@ -1003,6 +1003,57 @@ typedef struct {
     int tls_aad_len;
 } S390X_AES_GCM_CTX;
 
+typedef struct {
+    union {
+        double align;
+        /*-
+         * Padding is chosen so that ccm.kmac_param.k overlaps with key.k and
+         * ccm.fc with key.k.rounds. Remember that on s390x, an AES_KEY's
+         * rounds field is used to store the function code and that the key
+         * schedule is not stored (if aes hardware support is detected).
+         */
+        struct {
+            unsigned char pad[16];
+            AES_KEY k;
+        } key;
+
+        struct {
+           /*-
+            * KMAC-AES parameter block
+            * (see z/Architecture Principles of Operation SA22-7832-11)
+            */
+            struct {
+                union {
+                    unsigned long long g[2];
+                    unsigned char b[16];
+                } icv;
+                unsigned char k[32];
+            } kmac_param;
+
+            union {
+                unsigned long long g[2];
+                unsigned char b[16];
+            } nonce;
+            union {
+                unsigned long long g[2];
+                unsigned char b[16];
+            } buf;
+
```

Comment:
But fields below are not significant to instruction in question. Is it correctly understood? That only above fields are? In such case a commentary would be more than appropriate :-)

Reply:
Id say comment placed *inside* ccm struct indicates that it refers to the kmac_param field below rather than to whole struct. If you say that other conventions are common, ill add a comment.

Reply:
Ah! It looks like I've misplaced my comment. kmac_param is the one specific to kmac instruction and already nonce is something that kmac pays no attention to, but something that is used internally. And that's what I'm trying to say, I'd appreciate a note that delimits fields are not to be looked for in "principles of operation".

Reply:
POP reference is placed directly above kmac_param struct and delimiter is \n directly below kmac_param struct. I mean i could place comment inside kmac_param struct to indicate it only applies to fields of kmac_param. But youd like to see a comment below kmac_param which says something along the lines of "end of kmac param block" ?

Reply:
I do recognize that one can interpret commentary placement as hint, but I argue that that it's too subtle. In other words the answer to question is yes, I'd argue that explicit commentary delimiter is appropriate.

[On side note one can even wonder if it would be appropriate to clarify why kmac_param.k is chosen to be 32 bytes. ;-)]

Reply:
> 32 byte key field is used for all key sizes because

*I* understand why 32 bytes is used. Question was rather how hard would it be for somebody who is reading code for the first time to figure it out. That's what commentaries are about, about somebody else :-)